qa/ingest-with-mustache/src/test/java/org/elasticsearch/ingest/IngestDocumentMustacheIT.java

```diff
@@ -44,6 +44,30 @@ public void testAccessMetaDataViaTemplate() {
         assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("2 bar"));
     }
 
+    // TODO(talevy): expected?
+    public void testAccessSubFieldsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooDoc = new HashMap<>();
+        fooDoc.put("bar", "value");
+        document.put("foo", fooDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
```

Comment:
I think this expected behaviour. Mustache is just walking the map and finds `value` here, which then gets merged with "1 ".